include/myst/thread.h

```diff
@@ -405,13 +438,66 @@ MYST_INLINE int myst_thread_queue_remove_thread(
     return found ? pos : -1;
 }
 
+/** Find and remove up to `n` threads from the queue such that
+ * <waiting bitset> & <passed in bitset> != 0.
+ * Removed threads are placed in the `matches` queue.
+ */
+MYST_INLINE size_t myst_thread_queue_search_remove_bitset(
+    myst_thread_queue_t* queue,
+    myst_thread_queue_t* matches,
+    size_t n,
+    uint32_t bitset)
```

Comment:
I based the variable name off of the operation name `FUTEX_WAIT_BITSET`. Do you think bitmask would provide more clarity?

Reply:
According to https://man7.org/linux/man-pages/man2/futex.2.html, `BITSET` refers to an operation of setting a bit mask.

> FUTEX_WAIT_BITSET (since Linux 2.6.25)
              This operation is like FUTEX_WAIT except that val3 is used
              to provide a 32-bit bit mask to the kernel.

Reply:
I think bitset in this context means "set of bits" rather than "setting bits". For example, C++ has data structure called [bitset](https://www.cplusplus.com/reference/bitset/bitset/). I would prefer to keep the name bitset as it also aligns with the operation name.

Reply:
I prefer bitset because it matches the Linux definition.

Reply:
Also, Linux itself uses the name `bitset` within the kernel in the futex implementation.